### ISSUE TITLE
Print feature hilighted after using the search functionnality (lizmap_search)

### ISSUE THREAD

**Issue by arno974** [3liz/lizmap-web-client#1995]
Currently, it is not possible to print a map with the highlighted feature after using the search feature.

In the case of using the database search functionality (lizmap_search) it might be interesting to allow the user to print the searched entity.

I guess, that would require several changes :
- change lizmap_search materialized_view to add a primary key (see code below)
- add lizmap_search layer to the qgis project
- Update setSql function from lizmapFts.class.php to add  id to the SQL query (SELECT id, item_layer, item_label)
- Update getData from lizmapFts.class.php to add the id ($data[$key]['features'][] = array('id'=> $item->id,)
- Update search.js to add the id to the attributes (see code below)
- Add a custom js to specify the query (request: 'GETSELECTIONTOKEN')
 
_Code Javascript :_ 
Dans function addSearch( )
`text += '<li><a href="#'+bbox.toBBOX()+'" data="'+ftsGeometry.toString()+'" id="search-'+ ftsFeat.id +'">'+lab+'</a></li>';`

Dans function updateExternalSearch( aHTML )
```
var _featId = $(this).attr('id').replace('search-', '');
feat = new OpenLayers.Feature.Vector(geom, {id : _featId});
```

_Code SQL_ 
```
CREATE MATERIALIZED VIEW public.lizmap_search
TABLESPACE pg_default
AS 
	SELECT row_number() over () as id,
	*
from (
       ( 
            SELECT 'mon_text'::text AS item_layer,
	    ma_couche.nom AS item_label,
	    NULL::text AS item_filter,
	    NULL::text AS item_project,
	    communes.geom
	   FROM ma_table
       ) 
       UNION ALL ...
) as u
WITH DATA;
```

**Comment by mdouchin** [comment]
Or we could simply use the redilining capability of QGIS Server, as it is done since Lizmap 3.4 to print the geometries added with the new Draw feature cc @nboisteault

**Comment by arno974** [comment]
Yes indeed it will require less changes as the geometry is already return and available in the data of the href.

**Comment by arno974** [comment]
Hi,

As I really need this functionality I've made some changes in the core of my lizmap code.
Inside the map.js file I've added (at [line 4054](https://github.com/3liz/lizmap-web-client/blob/220d373c778053da841f041f4cd38063018a3faf/lizmap/www/assets/js/map.js#L4054)) :

`
     
      var locateLayer = map.getLayersByName('locatelayer');
      if (locateLayer.length != 0) {
        locateLayer = locateLayer[0];
        if (locateLayer.features.length != 0) {
          var selectedFeat = locateLayer.features[0];	
          var wkt = new OpenLayers.Format.WKT; 
          var style = '<?xml version="1.0" encoding="UTF-8"?><StyledLayerDescriptor xmlns="http://www.opengis.net/sld" xmlns:ogc="http://www.opengis.net/ogc" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.1.0" xmlns:xlink="http://www.w3.org/1999/xlink" xsi:schemaLocation="http://www.opengis.net/sld http://schemas.opengis.net/sld/1.1.0/StyledLayerDescriptor.xsd"   xmlns:se="http://www.opengis.net/se"> <UserStyle>   <FeatureTypeStyle><Rule><PolygonSymbolizer><Stroke><SvgParameter name="stroke">#ffff33</SvgParameter><SvgParameter name="stroke-opacity">1</SvgParameter><SvgParameter name="stroke-width">3</SvgParameter></Stroke><Fill><SvgParameter name="fill">#ffff33</SvgParameter><SvgParameter name="fill-opacity">0</SvgParameter></Fill></PolygonSymbolizer></Rule></FeatureTypeStyle></UserStyle></StyledLayerDescriptor>;<?xml version="1.0" encoding="UTF-8"?><StyledLayerDescriptor xmlns="http://www.opengis.net/sld"   xmlns:ogc="http://www.opengis.net/ogc" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.1.0" xmlns:xlink="http://www.w3.org/1999/xlink" xsi:schemaLocation="http://www.opengis.net/sld http://schemas.opengis.net/sld/1.1.0/StyledLayerDescriptor.xsd" xmlns:se="http://www.opengis.net/se"><UserStyle><FeatureTypeStyle><Rule><PolygonSymbolizer><Stroke><SvgParameter name="stroke">#ffff33</SvgParameter><SvgParameter name="stroke-opacity">1</SvgParameter><SvgParameter name="stroke-width">3</SvgParameter></Stroke><Fill><SvgParameter name="fill">#ffff33</SvgParameter><SvgParameter name="fill-opacity">0</SvgParameter></Fill></PolygonSymbolizer></Rule></FeatureTypeStyle></UserStyle></StyledLayerDescriptor>';
			    
          if (printInProjectProjection && lizMap.config.options.qgisProjectProjection.ref != lizMap.config.options.projection.ref) {
            // If we print in the QGIS project projection
            selectedFeat.geometry.transform(map.projection, project_projection); 
          }         
          printParams[dragCtrl.layout.mapId+':HIGHLIGHT_GEOM'] =  wkt.write(selectedFeat);
          printParams[dragCtrl.layout.mapId+':HIGHLIGHT_SYMBOL'] = style;
        }
      }
`

Do you think it would be possible to have this functionality in the next lizmap version ?

**Comment by arno974** [comment]
By the way, it might be interesting to expose the printParams[dragCtrl.layout.mapId+':HIGHLIGHT_GEOM'] in the lizMap variable. So we could simply access to it inside custom JS code for example. Is it possible ?

**Comment by arno974** [comment]
Hi guys,

Any chance to have this feature integrated ?

Thanks
